pkg/controllers/allocation/batch.go

```diff
@@ -0,0 +1,178 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package allocation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	opAdd  = "add"
+	opWait = "wait"
+)
+
+// Batcher is a batch manager for multiple objects
+type Batcher struct {
+	// MaxBatchPeriod is the maximum amount of time to batch incoming pods before flushing
+	MaxBatchPeriod time.Duration
```

Comment:
Consider just calling this MaxPeriod since it's already contextualized by Batcher